### ISSUE TITLE
[SURE-9588] Issue setting folder S3 on etcd backups

### ISSUE THREAD

**Issue by aldelsa** [rancher/cluster-api-provider-rke2#499]
Hi,
We are trying to configure the etcd backup using S3 bucket parameters in the RKE2ControlPlane manifest:

```
  etcd:
      backupConfig:
        retention: "10"
        scheduleCron: "*/3 * * * *"
        s3:
          bucket: backups-rancher
          endpoint: test-rancher.com
          folder: backup
          enforceSslVerify: true
          s3CredentialSecret:
            apiVersion: v1
            kind: Secret
            namespace: rke2test
            name: creds-etcd-backup
```

The problem is the next part of the code: 

https://github.com/rancher/cluster-api-provider-rke2/blob/3d7df660eddfe972f8135eefb21b6411c9c29124/pkg/rke2/config.go#L322

If I use that s3 folder, I get "directory does not exist" message from rke2-server service because that folder does not exist on the machine for etcd backup, but it exists in my bucket.

The parameter ``folder`` should be independent from the ``EtcdSnapshotDir.``

Thanks

**Comment by aldelsa** [comment]
I found this PR #278 but I don't understand the reason of that.

**Comment by furkatgofurov7** [comment]
> I found this PR #278 @furkatgofurov7 but I don't understand the reason of that.

@aldelsa thanks for reporting the issue. I opened #549 which reverts the commit (the initial idea was to differentiate the backup location that is later consumed by other project, however it was not a correct assumption to change it the way it is now), PTAL and let me know if it solves your issue.

**Comment by kkaempf** [comment]
reopening and moving to "To Test"

**Comment by thehejik** [comment]
**Test report**
TL;DR version: 
The fixed RKE2 provider is working against AWS S3 as expected 

* rancher v2.10-head
* dev turtles (image and chart built from `4815395eecb7e5cb36e74a53bfec5e6c784110f0`)
* dev RKE2 provider (custom cp-rke2 and bootstrap-rke2 images built from `4815395eecb7e5cb36e74a53bfec5e6c784110f0`). To inject self made RKE2 provider image I used this new feature https://github.com/rancher/turtles/issues/761#issuecomment-2615781644.

---
Detailed version:
I've expanded `RKE2ControlPlane` resource by following `.etcd` block:
```yaml
apiVersion: controlplane.cluster.x-k8s.io/v1beta1
kind: RKE2ControlPlane
metadata:
  name: th-rke2-capv-cpi
  namespace: default
spec:
  serverConfig:
    etcd:
      backupConfig:
        retention: "10"
        scheduleCron: "*/3 * * * *"
        s3:
          bucket: thehejik-capi
          endpoint: s3.amazonaws.com
          region: eu-central-2  ### This is important otherwise the bucket is not reachable!
          folder: rke2-backup
          enforceSslVerify: true
          s3CredentialSecret:
            apiVersion: v1
            kind: Secret
            namespace: default
            name: s3-creds
...
```
The aws credentials were created on local rancher cluster by this command:
```bash
k create secret generic s3-creds --from-literal=aws_access_key_id=REDACTED --from-literal=aws_secret_access_key=REDACTED -n default
```

After successful capv+caprke2 downstream cluster provisioning the cp-rke2 provider created following config  `/etc/rancher/rke2/config.yaml` on RKE2 cp node:
```yaml
disable-cloud-controller: true
etcd-s3: true
etcd-s3-access-key: REDACTED
etcd-s3-bucket: thehejik-capi
etcd-s3-endpoint: s3.amazonaws.com
etcd-s3-folder: rke2-backup
etcd-s3-region: eu-central-2
etcd-s3-secret-key: REDACTED
etcd-snapshot-retention: "10"
etcd-snapshot-schedule-cron: '*/3 * * * *'
tls-san:
  - 10.124.139.10
cluster-cidr: 10.0.0.0/16
kubelet-arg:
  - --cloud-provider=external
token: REDACTED
```
And after successful provisioning the rke2 cluster automatically started to create etcd snapshots of the rke2 etcd server on AWS S3 in my bucket under given folder. The folder doesn't need to exist, the rke2 will create it automatically.

![Image](https://github.com/user-attachments/assets/a1587903-ab5c-4b90-9b7f-a12de93e8ea3)

Note: one can verify if the rke2 etcd setting is correct by performing `rke2 etcd-snapshot save --name bla` manually.

I didn't test etcd restore functionality as it is out of scope of this issue and should be tested by rke2 team already.
